src/OrleansProviders/Streams/Common/SimpleQueueAdapterCache.cs

```diff
@@ -54,5 +56,17 @@ public int Size
         {
             get { return caches.Select(pair => pair.Value.Size).Sum(); }
         }
+
+        public static int ParseSize(ReadOnlyDictionary<string, string> config, int defaultSize)
+        {
+            string cacheSizeString;
+            int cacheSize = defaultSize;
+            if (config.TryGetValue(CACHE_SIZE_PARAM, out cacheSizeString))
+            {
+                if (!int.TryParse(cacheSizeString, out cacheSize))
```

Comment:
Use `ProviderConfigurationExtensions.GetIntProperty()` instead of most of this code?